joomla-projects/gsoc18_adaptive_images
# Different focus selection for different widths

### Summary of Changes

Now, We can define different focus area for different display width (optionally). 

- If in case the user has not defined the focus for a particular width, then the maximum of all the focus areas has been chosen and shown.

- In case, User only defines the focus area for one width only, It would be considered as the focus area of all the possible width.

### Testing Instructions

- In media-manager upload any image.
- Go to edit option of that image, goto Smart Crop tab.
-  Select a focus area for the given width.
-  Select another focus area by 
	- Changeing the width value from the combo box
	- Then selecting the focus for that width.
- After that Save and Close the edit tab.
- Now add that image to the article and see If the image changes when the browser window width is changed.

Note : Please check if following plugins are enabled or not (If not please enable them)
1. adaptiveimage (Content Plugin)
2. smartcrop (Media-Action Plugin)


### Expected result

The changes that the user may observe.
1. The images changes at the front-end when the width of the view port (window size) is being changed.
2. The cropped images can be found in ```/images/.cache```.

### Screenshots
#### Selecting the focus area for diffrent width of the view port.
![screenshot 16](https://user-images.githubusercontent.com/18314265/42416668-285605c2-8293-11e8-9425-872a55092cc1.png)
![screenshot 17](https://user-images.githubusercontent.com/18314265/42416669-28ba1c24-8293-11e8-9da4-fb4e8f107c96.png)
![screenshot 18](https://user-images.githubusercontent.com/18314265/42416670-28f87078-8293-11e8-9563-55db8ace6fdf.png)
![screenshot 19](https://user-images.githubusercontent.com/18314265/42416671-29359b2e-8293-11e8-887c-7bd1f59a3c18.png)
#### Cropped images for diffrent width
![screenshot 20](https://user-images.githubusercontent.com/18314265/42416676-4c51266e-8293-11e8-90d9-81bbfaca3b6b.png)
